src/traces/scattergl/index.js

```diff
@@ -302,6 +302,8 @@ function sceneUpdate(gd, subplot) {
             }
             if(scene.scatter2d) {
                 clearViewport(scene.scatter2d, vp);
+            } else if(scene.line2d) {
+                clearViewport(scene.line2d, vp);
```

Comment:
The first arg to `clearViewport` is just used to find the `regl._gl` context... and so I guess the reason `select2d` is off by itself is that select gets its own context separate from the main data context?

Might be clearer then to refactor this as something like
```js
var anyComponent = scene.scatter2d || scene.line2d || (scene.glText || [])[0];
if(anyComponent) clearViewport(anyComponent, vp);
```

Which would also make it easier to add `error2d` (can that ever be the only thing drawn?) and `fill2d` to the lookup list, looks plausible that these also have bugs similar to #2888.

BTW, looking at what `clearViewport` does - ie clears a rectangle from the context - what happens if two polar subplots have overlapping bounding boxes? Like if you made a hexagonal packing of polar subplots or you do something funny with sectors, like you make one large pac-man subplot eating a smaller one?

Reply:
> what happens if two polar subplots have overlapping bounding boxes?

This happens -> https://github.com/plotly/plotly.js/issues/2797

Reply:
> Might be clearer then to refactor this as something like

Absolutely -> c3044e094eacfc25fbd2f85648c66c4bf8693994

> add error2d (can that ever be the only thing drawn?)

Traces with error bar will always have either an active `scene.scatter2d` or `scene.line2d`.


> BTW, looking at what clearViewport does - ie clears a rectangle from the context -

We should take a look at how much of a perf gain is to clear only a fraction of the canvas on  pan/select. If that's negligible, then I'm thinking we should be keeping track of a graph-wide regl-draw-queue instead, always clear the whole canvas and draw all regl parts on pan/select. This should make this a lot more manageable. cc @dy

Reply:
> Traces with error bar will always have either an active `scene.scatter2d` or `scene.line2d`.

Even if `mode: 'none'`? That works fine for `scatter`, and sometimes tries to work with `scattergl` #2900

`fill` works with `mode: 'none'` too.

Reply:
Bulletproofed in https://github.com/plotly/plotly.js/pull/2895/commits/75786e40279e1cfbbc04333be7fb0d15d2f3be39